src/Expander/Error.hs

```diff
@@ -60,6 +62,24 @@ data ExpansionErr
   | CircularImports ModuleName [ModuleName]
   deriving (Show)
 
+-- | A newtype to add a type level witness that differentiates between @Actual@
+-- and @Expected@ @SyntacticCategory@
+newtype Expected a = Expected { unExpected :: a }
+  deriving newtype Show
+
+-- | helper function to construct an @Expected@
+expected :: a -> Expected a
+expected = Expected
```

Comment:
What's the purpose of this wrapper function? It looks like a smart constructor, but it doesn't check any condition, so I don't see why callers should use `expected` instead of `Expected`.

Reply:
Ah yes I forgot to mention this. The purpose of the helper function is to separate the constructor from the exported API. So should `Expected` or `Actual` ever change then all their call sites won't need to change because from the beginning of their existence we define and treat `expected` and `actual` as _the only way_ to create an `Actual` or `Expected`.

I felt that this was more in line with david's original suggestion because if we took this to a logical next step and started to `parse, don't validate` then these helpers would be the site to do the parsing since they are the functions that add the type witnesses to the type level.

We can totally remove them if you like. If it were just me I would hide the `Actual` and `Expected` constructors and only export the `actual` and `expected` functions to enforce the aforementioned API.

Reply:
Since Klister is an executable, not a library, I don't think it's worth jumping through hoops to keep the code forward- or backward-compatible, as there aren't that many call sites to update. But now that you've jumped through the hoop, I don't want to make you jump backwards through the same hoop either! It's fine either way. If you do want to keep the function wrappers though, let's go all the way and not export the data constructors, like you suggest.

I am a fan of the parse-don't-validate approach when it makes sense, but I don't see how that approach applies here, because there are no values which would fail to parse. The purpose of the newtype is simply to tag the two arguments so that the code looks wrong if the two arguments are swapped.

By the way, `requireDeclarationCat` is already using the parse-don't-validate approach, in that it parses a `MacroDest` into a `(DeclTreePtr, DeclOutputScopesPtr)`.